src/rdkafka.c

```diff
@@ -1911,6 +1911,9 @@ rd_kafka_query_watermark_offsets (rd_kafka_t *rk, const char *topic,
 		return RD_KAFKA_RESP_ERR__UNKNOWN_PARTITION;
 	rktp = rd_kafka_toppar_s2i(s_rktp);
 
+	/* Query for the topic leader (async) */
```

Comment:
We only need to do this if there is no leader found, so maybe move this down a couple of lines below the failing rd_kafka_toppar_leader() check?